src/components/AnonymousChat/MessageChannelItem.tsx

```diff
@@ -1,26 +1,68 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+// eslint-disable-next-line no-use-before-define
 import * as React from 'react';
 
 import BaseChannelItem from './BaseChannelItem';
+import useChatUtilsHook from '../../hooks/core/chat/useChatUtilsHook';
+import useProfileHook from '../../hooks/core/profile/useProfileHook';
 import {BaseChannelItemTypeProps} from '../../../types/component/AnonymousChat/BaseChannelItem.types';
+import {ChannelList} from '../../../types/database/schema/ChannelList.types';
+import {Context} from '../../context';
 import {MessageChannelItemProps} from '../../../types/component/AnonymousChat/MessageChannelItem.types';
 import {calculateTime} from '../../utils/time';
+import {getChatName} from '../../utils/string/StringUtils';
 
 const MessageChannelItem: (props: MessageChannelItemProps) => React.ReactElement = ({
   item,
-  onChannelPressed
+  onChannelPressed,
+  hasFollowButton = false,
+  handleFollow
 }) => {
+  const {signedProfileId} = useProfileHook();
+
+  const determineChatType = (data: ChannelList) => {
+    const isAnonymous = Boolean(data?.channelType === 'ANON_PM');
+    if (!isAnonymous) return BaseChannelItemTypeProps.SIGNED_PM;
+    return BaseChannelItemTypeProps.ANON_PM;
+  };
+
+  const checkIsMe = (data: ChannelList, type: BaseChannelItemTypeProps): boolean => {
+    const rawJson = data?.rawJson;
+    const lastMessage = rawJson?.firstMessage ?? rawJson?.message;
+    const isSystemMessage = lastMessage?.isSystem || lastMessage?.type === 'system';
+
+    if (type === BaseChannelItemTypeProps.SIGNED_PM) {
+      if (isSystemMessage) return false;
+      return lastMessage?.user?.id === signedProfileId;
+    }
+    return item?.user?.isMe ?? false;
+  };
+
+  const {handleTextSystem} = useChatUtilsHook();
+  const [profile] = (React.useContext(Context) as unknown as any).profile;
+  const type = determineChatType(item);
+  const isMe = checkIsMe(item, type);
+
   return (
     <BaseChannelItem
-      type={BaseChannelItemTypeProps.ANON_PM}
-      message={item?.description || ' '}
-      name={item?.name}
+      type={type}
+      message={handleTextSystem(item)}
+      name={getChatName(item?.name, profile?.myProfile?.username)}
       picture={item?.channelPicture}
+      postMaker={item?.rawJson}
       time={calculateTime(item?.lastUpdatedAt, true)}
       onPress={onChannelPressed}
       unreadCount={item?.unreadCount}
-      isMe={item?.user?.isMe}
+      isMe={isMe}
+      hasFollowButton={hasFollowButton}
+      handleFollow={handleFollow}
+      channelType={item?.channelType}
     />
   );
 };
 
-export default MessageChannelItem;
+const isEqual = (prevProps, nextProps) => {
+  return nextProps.item === prevProps.item;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `isEqual` function only checks if the `item` prop is equal between the previous and next props. If other props change but `item` remains the same, the component will not re-render, which might not be the desired behavior. Consider using a deep equality check or include other props in the comparison if they should trigger a re-render.

68:
Using `React.memo` can optimize your component by avoiding unnecessary renders, but it's not always necessary and could even lead to performance issues if used incorrectly. Make sure that this optimization is actually needed here.


<!-- This is an auto-generated comment by OSS CodeRabbit -->